### PR TITLE
boards: add support for QT Py RP2040

### PR DESCRIPTION
**Pull request by hlord2000**
Adds support for the Adafruit QT Py RP2040.

### CODE REVIEW

File: boards/arm/adafruit_qt_py_rp2040/adafruit_qt_py_rp2040.dts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	status = "okay";
+	pinctrl-0 = <&adc_default>;
+	pinctrl-names = "default";
+};
+
+&pio0 {
+	status = "okay";
+};
+
+&usbd {
```
Comment (soburi) [inline boards/arm/adafruit_qt_py_rp2040/adafruit_qt_py_rp2040.dts:110]: In generally, usbd has set zephyr_udc0 label.
It seems better to follow that custom.

**Review by soburi — CHANGES_REQUESTED** [review]

**Review by yonsch — APPROVED** [review]
LGTM, @hlord2000 can you fix the request made by @soburi ?

**Comment by raveious** [comment]
bump!

**Comment by raveious** [comment]
So I was just about to open a PR for the same board when I found this PR. I pulled change and it seems to work just fine with the samples I tried out on it.

Is there any reason why USB CDC ACM is setup as the console? A lot of the RP2040 boards have a UART as the console, and this board seems to be able to do it too.

Is there anything I can do to help get this merged?

**Comment by raveious** [comment]
Ok. I got bored. I pulled this branch and did all the actions. If this PR doesn't get figured out, I can open a new one with everything done.

**Comment by hlord2000** [comment]
> So I was just about to open a PR for the same board when I found this PR. I pulled change and it seems to work just fine with the samples I tried out on it.
> 
> 
> 
> Is there any reason why USB CDC ACM is setup as the console? A lot of the RP2040 boards have a UART as the console, and this board seems to be able to do it too.
> 
> 
> 
> @hlord2000 @soburi Is there anything I can do to help get this merged?

#I will update based on your fork in a bit. I chose USB CDC ACM for this board since it's what circuitpython etc use on xiao form factor boards, may be best to switch back.

**Comment by raveious** [comment]
> > So I was just about to open a PR for the same board when I found this PR. I pulled change and it seems to work just fine with the samples I tried out on it.
> > Is there any reason why USB CDC ACM is setup as the console? A lot of the RP2040 boards have a UART as the console, and this board seems to be able to do it too.
> > @hlord2000 @soburi Is there anything I can do to help get this merged?
> 
> #I will update based on your fork in a bit. I chose USB CDC ACM for this board since it's what circuitpython etc use on xiao form factor boards, may be best to switch back.

I have everything done. Want me to just make a new PR?

**Comment by hlord2000** [comment]
> > > So I was just about to open a PR for the same board when I found this PR. I pulled change and it seems to work just fine with the samples I tried out on it.
> 
> > > Is there any reason why USB CDC ACM is setup as the console? A lot of the RP2040 boards have a UART as the console, and this board seems to be able to do it too.
> 
> > > @hlord2000 @soburi Is there anything I can do to help get this merged?
> 
> > 
> 
> > #I will update based on your fork in a bit. I chose USB CDC ACM for this board since it's what circuitpython etc use on xiao form factor boards, may be best to switch back.
> 
> 
> 
> I have everything done. Want me to just make a new PR?

Sure! Can you comment the PR link so I can check it out too? Thanks for fixing things.

**Comment by raveious** [comment]
> > > > So I was just about to open a PR for the same board when I found this PR. I pulled change and it seems to work just fine with the samples I tried out on it.
> > 
> > 
> > > > Is there any reason why USB CDC ACM is setup as the console? A lot of the RP2040 boards have a UART as the console, and this board seems to be able to do it too.
> > 
> > 
> > > > @hlord2000 @soburi Is there anything I can do to help get this merged?
> > 
> > 
> > > 
> > 
> > 
> > > #I will update based on your fork in a bit. I chose USB CDC ACM for this board since it's what circuitpython etc use on xiao form factor boards, may be best to switch back.
> > 
> > 
> > I have everything done. Want me to just make a new PR?
> 
> Sure! Can you comment the PR link so I can check it out too? Thanks for fixing things.

Take a look at #66603

**Comment by raveious** [comment]
Other PR merged, so this can be closed now.

**Pull request closed without merging**